src/libraries/System.Private.CoreLib/src/System/Type.cs

```diff
@@ -225,7 +225,15 @@ public virtual Type[] GetGenericParameterConstraints()
         [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)]
         public virtual MemberInfo[] GetMember(string name, MemberTypes type, BindingFlags bindingAttr) => throw new NotSupportedException(SR.NotSupported_SubclassOverride);
 
-        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)]
+        [DynamicallyAccessedMembers(
+            DynamicallyAccessedMemberTypes.PublicFields
+            | DynamicallyAccessedMemberTypes.PublicMethods
+            | DynamicallyAccessedMemberTypes.PublicEvents
+            | DynamicallyAccessedMemberTypes.PublicProperties
+            | DynamicallyAccessedMemberTypes.PublicConstructors
+            | DynamicallyAccessedMemberTypes.PublicNestedTypes)]
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2085:UnrecognizedReflectionPattern",
+            Justification = "Linker doesn't recognize GetMembers(BindingFlags.Public) but this is what the body is doing")]
```

Comment:
Should it?

Reply:
I've added the scenario to https://github.com/mono/linker/issues/1724#issuecomment-758209013. It is in the same vein.